pkgs/by-name/te/television/package.nix

```diff
@@ -1,60 +1,96 @@
   television,
   nix-update-script,
-  extraPackages ? [ ],
-}:
-rustPlatform.buildRustPackage (finalAttrs: {
-  pname = "television";
-  version = "0.13.2";
-
-  src = fetchFromGitHub {
-    owner = "alexpasmantier";
-    repo = "television";
-    tag = finalAttrs.version;
-    hash = "sha256-Ur6UTd3XsI2ZyVboQA9r3WDkl7hd1wQ0NCgTlYFF/C0=";
-  };
-
-  cargoHash = "sha256-LfaYRrJ4ZXoNVDsI650t+A7mWB9+2+znATp+mqDwTiE=";
-
-  nativeBuildInputs = [ makeWrapper ];
-
-  postInstall = lib.optionalString (extraPackages != [ ]) ''
-    wrapProgram $out/bin/tv \
-      --prefix PATH : ${lib.makeBinPath extraPackages}
-  '';
-
-  # TODO(@getchoo): Investigate selectively disabling some tests, or fixing them
-  # https://github.com/NixOS/nixpkgs/pull/423662#issuecomment-3156362941
-  doCheck = false;
-
-  passthru = {
-    tests.version = testers.testVersion {
-      package = television;
-      command = "XDG_DATA_HOME=$TMPDIR tv --version";
+  testers,
+  ...
+}@args:
+if args ? extraPackages then
+  lib.warn
+    "Overriding television with the 'extraPackages' attribute is deprecated. Please use `television.withPkgs (p: [ p.fd ...])` instead."
+    television.withPkgs
+    (_: args.extraPackages)
+else
+  rustPlatform.buildRustPackage (finalAttrs: {
+    pname = "television";
+    version = "0.13.2";
+
+    src = fetchFromGitHub {
+      owner = "alexpasmantier";
+      repo = "television";
+      tag = finalAttrs.version;
+      hash = "sha256-Ur6UTd3XsI2ZyVboQA9r3WDkl7hd1wQ0NCgTlYFF/C0=";
     };
-    updateScript = nix-update-script { };
-  };
-
-  meta = {
-    description = "Blazingly fast general purpose fuzzy finder TUI";
-    longDescription = ''
-      Television is a fast and versatile fuzzy finder TUI.
-      It lets you quickly search through any kind of data source (files, git
-      repositories, environment variables, docker images, you name it) using a
-      fuzzy matching algorithm and is designed to be easily extensible.
-    '';
-    homepage = "https://github.com/alexpasmantier/television";
-    changelog = "https://github.com/alexpasmantier/television/releases/tag/${finalAttrs.version}";
-    license = lib.licenses.mit;
-    mainProgram = "tv";
-    maintainers = with lib.maintainers; [
-      louis-thevenet
-      getchoo
+
+    outputs = [
+      "out"
+      "share"
     ];
-  };
-})
+
+    strictDeps = true;
+    nativeBuildInputs = [
+      installShellFiles
+      makeBinaryWrapper
+    ];
+
+    cargoHash = "sha256-LfaYRrJ4ZXoNVDsI650t+A7mWB9+2+znATp+mqDwTiE=";
+
+    doCheck = false;
+
+    # TODO: Add nushell completion once #428168 hits unstable
+    postInstall = ''
+      installManPage target/${stdenv.hostPlatform.rust.cargoShortTarget}/assets/tv.1
+
+      installShellCompletion --cmd tv \
+        television/utils/shell/completion.bash \
+        television/utils/shell/completion.fish \
+        television/utils/shell/completion.zsh
+
+      mkdir -p "$share/share"
+      cp -r cable/unix "$share/share"
```

Comment:
But I'm not really sure why we're copying these and overriding the cable directory at all. I can only really assume this is a reproducibility benefit? But I don't think that'd be worthwile

As far as I'm aware, out of band updates to channels are very much the intention upstream. I don't know what real-world benefit we'd get from vendoring channels this way when compatibility mechanisms are in place internally, and I'm not aware of a single breakage resulting from this. I'm also fairly sure this would outright break setting `--cable-dir` manually, as well as prevent any custom channels from being loaded at all really

Reply:
Yeah it does break cable-dir, I just tested. If out-of-band updates are the intention then I'll just remove this.

Reply:
I'll submit a patch upstream to see if we can get a "default" cable dir via envars. The main motivation I had for vendoring it wasn't reproducibility, but ensuring that television "just works" when a user runs something like `, tv files`, and not having to download the cables first.

Reply:
> I'll submit a patch upstream to see if we can get a "default" cable dir via envars.

I'm still not sure that's a good idea, as that would continue to prevent user-specific channels from being used OOTB without modifying the wrapper in some way (in this case, removing the env var or setting `--cable-dir` explicitly)

Ideally it'd be something a bit more advanced, like falling back on "global" channels installed with the package when none are present in `~/.config`. This would also probably benefit other distros and users of them